crates/sequencer/src/state.rs

```diff
@@ -143,6 +170,27 @@ impl BlockifierStateReader for &mut State {
     }
 }
 
+impl From<&SerializableState> for State {
```

Comment:
Why are we implementing the trait on a reference?

Reply:
So that we want to create a `State` from `SerializableState`, we can do so without moving the value and rather than borrowing it.

Reply:
I don't really see the need for this, specially as the implementation itself just copies everything? So in the end we are copying values

Reply:
But if you will do it directly on type `SerializableState`, then this copying will happen twice right?

Firstly you have to call this method on a clone, and then there will another copying inside the implementation. I think the reference approach is hence more optimizied.

Lemme know what you think.